web/web.go

```diff
@@ -0,0 +1,799 @@
+			tempColDef := sp.ColDefs[newColName]
+			tempColDef.T = ty
+			sp.ColDefs[newColName] = tempColDef
+		}
+
+		if len(v.Constraint) > 0 {
+			for _, constraint := range v.Constraint {
+				switch constraint {
+				case "NOT NULL":
+					spColDef := sp.ColDefs[newColName]
+					spColDef.NotNull = true
+					sp.ColDefs[newColName] = spColDef
+				default:
+					fmt.Println("skip")
+				}
+			}
+		}
+
+		app.conv.SpSchema[table] = sp
+
+	}
+	app.conv.AddPrimaryKeys()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(app.conv)
+}
+
+func rateSchema(cols, warnings int64, missingPKey bool) string {
+	switch {
+	case cols == 0:
+		return "GRAY"
+	case warnings == 0 && !missingPKey:
+		return "GREEN"
+	case warnings == 0 && missingPKey:
+		return "BLUE"
+	case good(cols, warnings) && !missingPKey:
+		return "BLUE"
+	case good(cols, warnings) && missingPKey:
+		return "BLUE"
+	case ok(cols, warnings) && !missingPKey:
+		return "YELLOW"
+	case ok(cols, warnings) && missingPKey:
+		return "YELLOW"
+	case !missingPKey:
+		return "RED"
+	default:
+		return "RED"
+	}
+}
+func good(total, badCount int64) bool {
+	return badCount < total/20
+}
+
+func ok(total, badCount int64) bool {
+	return badCount < total/3
+}
+func getConversionRate(w http.ResponseWriter, r *http.Request) {
+	reports := internal.AnalyzeTables(app.conv, nil)
+	rate := make(map[string]string)
+	for _, t := range reports {
+		rate[t.SpTable] = rateSchema(t.Cols, t.Warnings, t.SyntheticPKey != "")
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(rate)
+}
+
+func getSchemaAndReportFile(w http.ResponseWriter, r *http.Request) {
+	ioHelper := &conversion.IOStreams{In: os.Stdin, Out: os.Stdout}
+	dbName := app.dbName
+	var err error
+	now := time.Now()
+	if dbName == "" {
+		dbName, err = conversion.GetDatabaseName(app.driver, now)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Can not create database name : %v", err), 500)
+		}
+	}
+
+	filePrefix := dbName + "."
+
+	if err != nil {
+		fmt.Printf("\nCan't get database name: %v\n", err)
+		panic(fmt.Errorf("can't get database name"))
+	}
+	reportFileName := filePrefix + "report.txt"
+	schemaFileName := filePrefix + "schema.txt"
+
+	response := make(map[string]string)
+	conversion.WriteSchemaFile(app.conv, now, schemaFileName, ioHelper.Out)
+	conversion.Report(app.driver, nil, ioHelper.BytesRead, "", app.conv, reportFileName, ioHelper.Out)
+	reportAbsPath, err := filepath.Abs(reportFileName)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Can not create absolute path : %v", err), 500)
+	}
+	schemaAbsPath, err := filepath.Abs(schemaFileName)
+	response["reportFilePath"] = reportAbsPath
+	response["schemaFilePath"] = schemaAbsPath
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(response)
+}
+
+type tableInterleaveStatus struct {
+	notPossible bool
+	parent      string
+	comment     string
+}
+
+func checkForInterleavedTables(w http.ResponseWriter, r *http.Request) {
```

Comment:
Usual go style guidelines are to keep functions to about a page (about 50 lines or so)
This is much longer -- are there good ways to split this up into a couple of functions?

Reply:
updated function, splitted into multiple smaller functions.